### ISSUE TITLE
Feature Requests

### ISSUE THREAD

**Issue by tschinz** [sectore/timr-tui#33]
My feature requests
1. Add Countdown setup with human readable time e.g. countdown to "22h10" or "19:30" or "8:30AM"
2. Countdown edit should be freely adjustable not just up to the initial value should go as high as 999h99m99s
4. Add to crates.io should be able to installed with cargo install timr (create release)

Thanks looks amazing

**Comment by sectore** [comment]
>  1. Add Countdown setup with human readable time e.g. countdown to "22h10" or "19:30" or "8:30AM"

^ Could you explain this feature request in more detail, pls?


> 2. Countdown edit should be freely adjustable not just up to the initial value should go as high as 999h99m99s

With latest changes `main` (f3a6e07) users are able to edit `Countdown` and `Pomodoro` (work/pause) as they wish. Maximum value: 99h59m59s ([`code`](https://github.com/sectore/timr/blob/main/src/widgets/clock.rs#L24-L26)). 

Note: Editing those values won't change initial values, but you can do this by defining those values before starting the app:

```sh
# change initial value of `Countdown` to `20min`.
cargo run -- --countdown 20:00

# change initial value of `Pomodoro`  `work` to `33min`.
cargo run -- --work 33:00

# change initial value of `Pomodoro` `pause` to `15min`.
cargo run -- --pause 15:00
```

Check [README's section "Args"](https://github.com/sectore/timr?tab=readme-ov-file#args-%EF%B8%8F) for options and time formats.



>  3. Add to crates.io should be able to installed with cargo install timr (create release)

See #34

**Comment by tschinz** [comment]
> Note: Editing those values won't change initial values, but you can do this by defining those values before starting the app

Yes I saw that you can set the initial countdown values. But once set to a value, it will be the maximum value for that program lifetime. It's not possible to set something higher. For me that value should be the start value but with (e)dit it should be possible to set ANY value.

for 1. you want either countdown 2h (as it is right now) or count down until a specific time is reached. For example it currently it is 9h00 and you want to countdown to 10h12 the countdown will be effectively 1h12m.
In short instead of settings a coundown duration you set the countdown to a future time from the current time.
BTW: https://docs.rs/humantime/latest/humantime/fn.parse_duration.html is nice to parse a duration text input.

I hope this helps feel free to reach out again.

**Comment by sectore** [comment]
Thx for more details! `1.` looks good to me, will implement it later.  

> It's not possible to set something higher. 

BTW: As mentioned earlier, f3a6e07 has fixed it ^.

**Comment by tschinz** [comment]
> BTW: As mentioned earlier, https://github.com/sectore/timr/commit/f3a6e073e2275a1394ff98c553af82d98b1265b6 has fixed it ^.

my bad perfect

> Thx for more details! 1. looks good to me, will implement it later.

looking forwards to it

Last feature request: for the countdown once 0 has reached it would be great if the timer continues with positive values similar to a Rocket countdown:
T-5, T-4, T-3, T-2, T-1, T0, T+1, T+2, T+3, ....

Since I counting a lot of presentations or exams it is nice to know how much over the initial time the presenters are: I implemented something similar in m liftoff app: https://github.com/tschinz/liftoff/tree/main/liftoff-egui

**Comment by sectore** [comment]
@tschinz Pls check latest changes in `main`, especially #45 and #42 .

IMO the [previous request]([here](https://github.com/sectore/timr-tui/issues/33#issuecomment-2567119172)) of adding another edit mode to enter a specific time does not make sense anymore. Since the local time is visible in footer now, an user is able to do a simple math to get the right value for a countdown.

**Comment by tschinz** [comment]
dear @sectore, thanks so much for the implementation.

1. The rocket countdown is perfect.
2. For me to define the countdown with a time still does make sense. Let me explain. It is 10:29:14 and I want to count to 11:10:00. I can calculate it, but it is cumbersome even with the current time displayed in the corner. A time entry would be better.

My implementation proposition would be with <- and -> (as in the pomodoro timer) change the entry method from countdown a duration to countdown to a certain time (within the same day).

But it is your tool it is up to you, for me also in its current version very much usable (and in service)

**Comment by sectore** [comment]
@tschinz With latest PR #49 users can enter a local time to set a countdown. Demo in PR description. Have fun! 

Thx again for your suggestions / feedback!
